src/Microsoft.AspNet.Razor/Generator/TagHelpers/EscapedTagHelperCodeGenerator.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.AspNet.Razor.Parser.SyntaxTree;
+
+namespace Microsoft.AspNet.Razor.Generator
+{
+    /// <summary>
+    /// A <see cref="BlockCodeGenerator"/> that is responsible for generating valid 
+    /// <see cref="Compiler.LiteralChunk"/>s from escaped tag helpers.
+    /// </summary>
+    public class EscapedTagHelperCodeGenerator : SpanCodeGenerator
```

Comment:
not a big fan of this name.  makes it seem like we go down a different generation code path than any other element that doesn't match a tag helper descriptor (albeit after correcting the tag name)?  but this class really does a quick fixup of the generated tree to correct the tag name.  suggest `RemoveBangCodeGenerator`.

more generally, why does the escaped tag name last this long?  couldn't you leave the `'!'` character out of the original `Span`, perhaps with an `Association` back to a `Span` that includes the `'!'` as well.